include/swift/Reflection/ReflectionContext.h

```diff
@@ -415,24 +424,28 @@ class ReflectionContext
 
     assert(SecIdx < SecHdrVec.size() && "malformed ELF object");
 
-    const typename T::Section *SecHdrStrTab = SecHdrVec[SecIdx];
+    std::unique_ptr<const typename T::Section> &SecHdrStrTab =
+        SecHdrVec[SecIdx];
     typename T::Offset StrTabOffset = SecHdrStrTab->sh_offset;
     typename T::Size StrTabSize = SecHdrStrTab->sh_size;
 
-    auto StrTabStart =
-        RemoteAddress(ImageStart.getAddressData() + StrTabOffset);
-    auto StrTabBuf = this->getReader().readBytes(StrTabStart, StrTabSize);
-    auto StrTab = reinterpret_cast<const char *>(StrTabBuf.get());
+    // The string table is also not mapped in the child process,
+    // read its contents from the file buffer.
+    uint64_t StrTabStart = FileBufferStartAddress + StrTabOffset;
+    std::vector<char> StrTab(StrTabSize);
+    memcpy(StrTab.data(), (void *)(StrTabStart), StrTabSize);
```

Comment:
Why do we need to copy the table here? Could we just access it in the buffer directly?

Reply:
You're right! We can.

Reply:
I just moved the copy from the child process to the buffer, didn't stop to think if it was actually needed.